swap_meet/vendor.py

```diff
@@ -1,2 +1,91 @@
+from .item import Item
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item 
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item 
+        else:
+            return False
+    
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == Item(category=category).category]
+    
+    def swap_items(self, Vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in Vendor.inventory:
+            Vendor.remove(their_item)
+            self.add(their_item)
+            Vendor.add(my_item)
+            self.remove(my_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, Vendor):
+        if len(self.inventory) > 0 and len(Vendor.inventory) > 0:
+            self.swap_items(Vendor, self.inventory[0], Vendor.inventory[0])
+            return True
+        else:
+            return False
+    
+    def get_best_in_category_by_attribute(self, category, attribute):
```

Comment:
I wonder if we should separate this out further. Maybe have helper functions for `min` and `max` to dry up the code a little bit!